Mathlib/Topology/Instances/EReal.lean

```diff
@@ -171,6 +170,154 @@ theorem tendsto_nhds_bot_iff_real {α : Type*} {m : α → EReal} {f : Filter α
   nhds_bot_basis.tendsto_right_iff.trans <| by simp only [true_implies, mem_Iio]
 #align ereal.tendsto_nhds_bot_iff_real EReal.tendsto_nhds_bot_iff_real
 
+/-! ### Liminfs and Limsups -/
+
+theorem liminf_le_liminf {α : Type*} {f : Filter α} {u v : α → EReal} (h : u ≤ᶠ[f] v) :
+    liminf u f ≤ liminf v f := Filter.liminf_le_liminf h
+
+theorem limsup_le_limsup {α : Type*} {f : Filter α} {u v : α → EReal} (h : u ≤ᶠ[f] v) :
+    limsup u f ≤ limsup v f := Filter.limsup_le_limsup h
+
+theorem limsup_add_le_lt₂ {α : Type*} {f : Filter α} {u v : α → EReal} {a b : EReal}
+    (ha : limsup u f < a) (hb : limsup v f < b) : limsup (u + v) f ≤ a + b := by
+  rcases eq_or_neBot f with (rfl | _); simp only [limsup_bot, bot_le]
+  rw [← @limsup_const EReal α _ f _ (a + b)]
+  apply limsup_le_limsup (Eventually.mp (Eventually.and (eventually_lt_of_limsup_lt ha)
+    (eventually_lt_of_limsup_lt hb)) (eventually_of_forall _))
+  simp only [Pi.add_apply, and_imp]
+  intro x
+  exact fun ux_lt_a vx_lt_b ↦ add_le_add (le_of_lt ux_lt_a) (le_of_lt vx_lt_b)
+
+theorem limsup_add_bot_of_ne_top {α : Type _} {f : Filter α} {u : α → EReal} {v : α → EReal}
+    (h : limsup u f = ⊥) (h' : limsup v f ≠ ⊤) : limsup (u + v) f = ⊥ := by
+  apply le_bot_iff.1
+  apply (le_iff_le_forall_real_gt ⊥ (limsup (u + v) f)).1
+  intro x
+  rcases exists_between_coe_real (h'.lt_top) with ⟨y, ⟨hy, _⟩⟩
+  rw [← sub_add_cancel x y, coe_add (x - y) y, coe_sub x y]
+  intro _
+  apply @limsup_add_le_lt₂ α f u v (x - y) y _ hy
+  rw [h, ← coe_sub x y]
+  exact bot_lt_coe (x - y)
+
+theorem limsup_add_le_add_limsup {α : Type _} {f : Filter α} {u v : α → EReal}
+    (h : limsup u f ≠ ⊥ ∨ limsup v f ≠ ⊤) (h' : limsup u f ≠ ⊤ ∨ limsup v f ≠ ⊥) :
+    limsup (u + v) f ≤ (limsup u f) + (limsup v f) := by
```

Comment:
Can you not employ some sort of `wlog` arugment here? (Maybe even a manual one using `suffices` instead of the actual `wlog`)

Reply:
Honestly, I am not sure how to do it in this case.